staff_features/resources/step_definitions/resource_tree_navigate.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+Given 'a Resource with two Archival Objects has been created' do
+  visit "#{STAFF_URL}/resources/new"
+
+  fill_in 'resource_title_', with: "Resource #{@uuid}"
+  fill_in 'resource_id_0_', with: "Resource #{@uuid}"
+  select 'Class', from: 'resource_level_'
+  element = find('#resource_lang_materials__0__language_and_script__language_')
+  element.send_keys('AU')
+  element.send_keys(:tab)
+
+  select 'Single', from: 'resource_dates__0__date_type_'
+  fill_in 'resource_dates__0__begin_', with: '2024'
+
+  fill_in 'resource_extents__0__number_', with: '10'
+  select 'Cassettes', from: 'resource_extents__0__extent_type_'
+
+  element = find('#resource_finding_aid_language_')
+  element.send_keys('ENG')
+  element.send_keys(:tab)
+
+  element = find('#resource_finding_aid_script_')
+  element.send_keys('Latin')
+  element.send_keys(:tab)
+
+  find('button', text: 'Save Resource', match: :first).click
+
+  wait_for_ajax
+  expect(page).to have_text "Resource Resource #{@uuid} created"
+
+  url_parts = current_url.split('/')
+  url_parts.pop
+  @resource_id = url_parts.pop
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 1 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 1 #{@uuid} on Resource Resource #{@uuid} created"
+
+  click_on 'Add Child'
+  wait_for_ajax
+  fill_in 'Title', with: "Archival Object 2 #{@uuid}"
+  select 'Class', from: 'Level of Description'
+  click_on 'Save'
+  expect(page).to have_text "Archival Object Archival Object 2 #{@uuid} created as child of Archival Object 1 #{@uuid} on Resource Resource #{@uuid}"
+  wait_for_ajax
+end
+
+Given 'the Resource is opened in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+When 'the user opens the Resource in edit mode' do
+  visit "#{STAFF_URL}/resources/#{@resource_id}/edit"
+
+  wait_for_ajax
+end
+
+Then 'the Resource is displayed as the top level of the navigation tree' do
+  rows = all('#tree-container .table .table-row', visible: :all)
+
+  expect(rows.length).to eq 4
+  expect(rows[0].text).to include "Resource #{@uuid}"
+end
+
+Then 'the tree includes the two Archival Objects' do
+  rows = all('#tree-container .table .table-row')
```

Comment:
just curious about why the visible: all filter wasn't used here but is in line 65

Reply:
If I recall correctly, I did it to try to check for all archival objects, even the hidden ones, but this is not the case for the resource tree. Nested archival objects are dynamically loaded on request. This step is not used anywhere. Removed.